### PR TITLE
Submit blocks from downloaded archive in batches

### PR DESCRIPTION
**Pull request by nazar-pc**
There are still things to optimize, but I think this might be enough with ~1600 blocks/s speed

### CODE REVIEW

File: backend/src/index.ts [hunk 1]
```diff
@@ -89,44 +137,55 @@ const processSourceBlocks = (target: Target) => async (source: Source) => {
...
+        for await (const [blocksBatch, lastBlockNumber] of readBlocksInBatches(archive, lastProcessedBlock)) {
+          if (lastTxPromise) {
+            await lastTxPromise;
           }
+          lastTxPromise = (async () => {
+            try {
+              const txHash = await target.sendBlocksBatchTx(feedId, signer, blocksBatch, nonce);
+              nonce++;
+
+              logger.debug(`Transaction included: ${polkadotAppsUrl}${txHash}`);
```
Comment (isSerge) [inline backend/src/index.ts:173]: this URL won't work for transactions, you can only query blocks

File: backend/src/target.ts [hunk 2]
```diff
@@ -76,6 +77,43 @@ class Target {
...
+        .batchAll(putCalls)
+        // it is required to specify nonce, otherwise transaction within same block will be rejected
+        // if nonce is -1 API will do the lookup for the right value
+        // https://polkadot.js.org/docs/api/cookbook/tx/#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce
+        .signAndSend(signer.address, { nonce: nonce || -1, signer }, (result) => {
+          if (result.isError) {
+            reject(result.status.toString());
+            unsub();
+          } else if (result.status.isInBlock) {
+            resolve(result.status.asInBlock);
```
Comment (isSerge) [inline backend/src/target.ts:104]: probably it will be more useful to resolve block hash

File: backend/src/index.ts (hunk 1 above)
Comment (nazar-pc) [reply backend/src/index.ts:173]: It is a bad variable name, turns out it is a block hash

File: backend/src/target.ts (hunk 2 above)
Comment (nazar-pc) [reply backend/src/target.ts:104]: This is a block hash, exactly the same code as in the regular version above

**Comment by nazar-pc** [comment]
Did some minor tweaks and cleanups

**Review by isSerge — APPROVED** [review]

**Pull request merged**